qiskit/transpiler/coupling.py

```diff
@@ -403,6 +398,44 @@ def largest_connected_component(self):
         """Return a set of qubits in the largest connected component."""
         return max(rx.weakly_connected_components(self.graph), key=len)
 
+    def components(self, respect_direction: bool = False) -> List["CouplingMap"]:
+        """Separate a CouplingMap into subgraph Coupling Maps for each connected component.
```

Comment:
I'm not 100% sure that we _shouldn't_ use graph terminology where appropriate.  The underlying structure is a graph, and I think anybody using these objects properly does actually need to understand that.  We can use it as well as quantum-specific terminology for sure, but I think the graph-based terminology is important to include because it's more precise.

Reply:
I'm not sure. I think the point of `CouplingMap` (and similar classes) is to provide a level of abstraction so users won't need to work directly with `CouplingMap.graph` unless they want to. I'm not saying that we shouldn't use graph terminology where necessary, only that it shouldn't be assumed to be known by users, so that where it is needed, we provide either a layman's description or link to a more detailed explanation.

Reply:
I updated the docstring in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/0936fb044b89de21cec8728c20519f5a6986c241 let me know if the updated documentation (which includes a simple example) is better now.